include/xtensor/xstrided_view.hpp

```diff
@@ -807,6 +807,66 @@ namespace xt
         );
     }
 
+    namespace detail
+    {
+        // if shape is signed do the check
+        template <class S, class R>
+        using do_shape_recalculation = std::
+            enable_if_t<std::is_signed<get_value_type_t<typename std::decay<S>::type>>::value, R>;
+
+        // if shape is unsigned pass through
+        template <class S, class R>
+        using no_shape_recalculation = std::
+            enable_if_t<!std::is_signed<get_value_type_t<typename std::decay<S>::type>>::value, R>;
+
+        template <typename T>
+        inline no_shape_recalculation<T, T> make_unsigned_shape(T shape)
+        {
+            return shape;
+        }
+
+        template <typename S, typename Enable = void>
+        struct rebind_shape;
+
+        template <class S>
+        struct rebind_shape<S, std::enable_if_t<!std::is_signed<get_value_type_t<typename std::decay_t<S>>>::value>>
+        {
+            using Shape = S;
+        };
+
+        template <class S>
+        struct rebind_shape<S, std::enable_if_t<std::is_signed<get_value_type_t<typename std::decay_t<S>>>::value>>
+        {
+            using Shape = rebind_container_t<size_t, S>;
+        };
+
+        template <class S, do_shape_recalculation<S, bool> = true>
+        inline auto recalculate_shape_impl(S& shape, size_t size)
+        {
+            using value_type = get_value_type_t<typename std::decay_t<S>>;
+            auto iter = std::find(shape.begin(), shape.end(), -1);
```

Comment:
You are missing a check on having just one `-1` I think. You should use `XTENSOR_ASSERT` for that